docs/developer/Configuration.md

```diff
@@ -0,0 +1,114 @@
+
+1. Notebook Specification: The most comprehensive form of configuration, defining what to render in a notebook, and how. It includes metadata, field definitions, forms, and views/sections. This is used for notebook-specific configurations and can be modified by users through the Designer application.
+
+2. App UI Theme: Global style customization for the mobile application, set through an environment variable. It affects all notebooks and is not user-changeable.
+
+3. Specific Configuration Switches: Environment variables that change global application behavior, applying to all notebooks and not changeable by users.
+
+The document provides guidance on when to use each type of configuration and includes a decision flowchart to help determine the appropriate configuration method for different scenarios.
+
+## FAIMS3 Configuration architecture
+
+Users define a notebook specification using the Designer application. This is combined with a colour and styling specification in the UI theme, along with special configuration switches, to deliver a mobile application with a specialised form.
+
+```mermaid
+graph TD
+    app[Mobile Application] --> form[Form Display]
+
+    designer[Designer] --> spec[Notebook Specification]
+    spec --> app
+
+    F[UI Theme] --> app
+    G[Configuration Switches] --> app
+
+    subgraph "Configuration"
+    spec
+    F
+    G
+    end
+
+    subgraph "User Interface"
+    app
+    form
+    end
+```
+
+## Notebook design / specification
+
+The most significant form of configuration in the application is the specification for the notebook. This defines **what** to render, and **how** it should be rendered. A notebook specification includes the following components, as shown in the diagram below
+
+- metadata: a set of key value pairs which contain
+  - important notebook metadata such as name, description, author etc
+  - configuration switches such as QR code behaviour (TODO document this further)
+  - arbitrary user metadata key-value pairs
+- field definitions: a set of fields with a unique name, this includes detailed specifications on how to render the field, it's allowable returned values etc
+- forms: top level forms which display as discrete forms in the application e.g. Site, Building - these are a collection of views, which are a collection of fields
+- views/sections: a sub-section within a form, which contain a collection of Fields identified from the field definitions list by their unique identifier
+
+**When should this be used**: The notebook specification is a powerful form of configuration for the behaviour of a _specific_ notebook. When the behaviour to be configured applies to a) a specific notebook b) individual fields/forms/sections c) some notebooks but not all d) should be configurable by the user within a given deployment, then the notebook specification should be used.
+
```

Comment:
Note that making a change to the notebook format should be done in a backward compatible way if possible - code should work if the option is missing or has an older style value.  Any change to the format will generally also require an update to the Designer to allow users to customise that option.